homeassistant/components/keenetic_ndms2/config_flow.py

```diff
@@ -88,6 +91,34 @@ async def async_step_import(self, user_input=None):
         """Import a config entry."""
         return await self.async_step_user(user_input)
 
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered device."""
+        # Filter out items not having "keenetic' in their name
+        if (
+            "keenetic"
+            not in discovery_info.get(ssdp.ATTR_UPNP_FRIENDLY_NAME, "").lower()
+        ):
+            return self.async_abort(reason="not_keenetic_ndms2")
+
+        host = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname
+
+        if self._already_configured(host):
+            return self.async_abort(reason="already_configured")
+
+        await self.async_set_unique_id(host)
```

Comment:
So after some thought, this should not be done..you could likely set the unique id based on the uuid provided in the announce, but then you need to make sure you also grab that during manual setup.

```suggestion
```

Reply:
As part of ssdp flow I need to set some unique ID for it so that there will be only one discovered config. Othervise the result is like this:
![Снимок экрана от 2021-03-04 08-54-24](https://user-images.githubusercontent.com/1551926/109899662-c5df9600-7cc8-11eb-80f5-9f615e82f8c6.png)

I think that's because the router broadcasts multiple services. And I'm not specifying any ST filter since I'm not sure which will be broadcasted with what set of the router firmware components (customizable).

gssdp-discover response: 
```
resource available
  USN:      uuid:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx::urn:schemas-upnp-org:device:WANConnectionDevice:1
  Location: http://192.168.0.1:49347/rootDesc.xml
resource available
  USN:      uuid:yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy::urn:schemas-upnp-org:device:InternetGatewayDevice:1
  Location: http://192.168.0.1:49347/rootDesc.xml
resource available
  USN:      uuid:yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy::upnp:rootdevice
  Location: http://192.168.0.1:49347/rootDesc.xml
resource available
  USN:      uuid:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
  Location: http://192.168.0.1:49347/rootDesc.xml
resource available
  USN:      uuid:zzzzzzzz-zzzz-zzzz-zzzz-zzzzzzzzzzzz
  Location: http://192.168.0.1:49347/rootDesc.xml
resource available
  USN:      uuid:yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy
  Location: http://192.168.0.1:49347/rootDesc.xml
resource available
  USN:      uuid:yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy::urn:schemas-upnp-org:service:Layer3Forwarding:1
  Location: http://192.168.0.1:49347/rootDesc.xml
resource available
  USN:      uuid:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx::urn:schemas-upnp-org:service:WANPPPConnection:1
  Location: http://192.168.0.1:49347/rootDesc.xml
resource available
  USN:      uuid:zzzzzzzz-zzzz-zzzz-zzzz-zzzzzzzzzzzz::urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1
  Location: http://192.168.0.1:49347/rootDesc.xml
resource available
  USN:      uuid:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx::urn:schemas-upnp-org:service:WANIPConnection:1
  Location: http://192.168.0.1:49347/rootDesc.xml
resource available
  USN:      uuid:zzzzzzzz-zzzz-zzzz-zzzz-zzzzzzzzzzzz::urn:schemas-upnp-org:device:WANDevice:1
  Location: http://192.168.0.1:49347/rootDesc.xml
```

Reply:
Another option for this would be 
```suggestion
        await self.async_set_unique_id(discovery_info.get(ssdp.ATTR_UPNP_UDN, host))
```

Reply:
Ignore discovery without UDN do not have fallback to hostname.

Preferably you would find the UDN even for non discovered entries. But that is a bit tricky.

Reply:
Updated to use UDN.
Obtaining the UDN on manual setup is kinda tricky and not always possible since: a - UPnP can be inactive on the router, b - router can be on the different networks so that multicast messages won't make it to HA.